libcontainer/setns_init_linux.go

```diff
@@ -33,11 +34,18 @@ func (l *linuxSetnsInit) Init() error {
 	runtime.LockOSThread()
 	defer runtime.UnlockOSThread()
 
+	// If config.json is modified after the container started, we should validate selinux again.
+	if l.config.ProcessLabel != "" && !selinux.GetEnabled() {
+		return fmt.Errorf("selinux label is specified in config, but selinux is disabled or not supported")
+	}
+
 	if !l.config.Config.NoNewKeyring {
-		if err := label.SetKeyLabel(l.config.ProcessLabel); err != nil {
-			return err
+		if l.config.ProcessLabel != "" {
+			if err := label.SetKeyLabel(l.config.ProcessLabel); err != nil {
```

Comment:
@rhatdan I think that all of the `Set*Label` commands should be a no-op if `l.config.ProcessLabel` is `""` -- so as to avoid these sorts of problems.

Reply:
Yes, it does make sense for don't check nil in definition of `label.Set*Label`, because we maybe need to delete the process label, so we should check `l.config.ProcessLabel` is `""` for all the `Set*Label` call in `runc`.